### DISCUSSION TITLE
Passing arguments to an async function in a callback

### DISCUSSION THREAD

**Discussion by Diegiwg** [zauberzeug/nicegui#348] [Show and tell]
For another project, I needed to call an asynchronous function as a button callback.

So, testing some alternatives, I got to the code shown in the video.

This is the most basic example, which can be expanded for more needs.

Example code: async_with_args.py

```py
import asyncio

class Aync:
    data = None

    def __init__(self, data):
        self.data = data

    async def run(self):
        print(f"Async {self.data}")
        await asyncio.sleep(1)
        print("End")

# Para utilizar a função como callback async, deve-se fazer: callback=Async("valor de teste").run
```

https://user-images.githubusercontent.com/15692310/217118877-2374f327-13a7-41b4-ab45-48cd33407b1e.mp4

**Comment by rodja** [comment]
That could be handy sometimes. Mostly we use `functools.partial` to acomplish execution of async functions with args:

```py
from nicegui import ui
from functools import partial

async def run(msg: str):
    await ui.run_javascript(f'alert("{msg}")')

ui.button('show "Hello"', on_click=partial(run, 'Hello'))

ui.run()
```

**Reply by Diegiwg** [reply]
I didn't know about functools, thanks a lot for the tip.

I'm going to use it on my systems now.
